### ISSUE TITLE
Key Error when sending messages from Room

### ISSUE THREAD

**Issue by sameersadiq1** [skoczen/will#306]
Bot responds fine when 1-1 private messaged but not in groups.

This is the error i see:

```
Wed, 15 Nov 2017 10:29:53 [CRITICAL] Error handling incoming event {'body': '@SameerSadiq can we try things with the bot now?  or do you still want us to wait?', 'xmpp_jid': u'237466_asktherunbot@conf.hipchat.com', 'from': '237466_asktherunbot@conf.hipchat.com/Steve Smith', 'parent': None, '_index': 0, 'from_jid': '237466_3081700@chat.hipchat.com', 'loaded_plugins': set([]), 'namespace': u'jabber:client', 'mid': '53a83932-513d-4493-adce-b99d39deea68', 'ts': '1510759558.902166', 'to': '237466_2745616@chat.hipchat.com/bot||proxy|pubproxy-c600.hipchat.com|5272', 'tag': u'{jabber:client}message', 'plugins': OrderedDict(), 'iterables': [], 'type': 'groupchat'}:
Traceback (most recent call last):
  File "/Users/Sameer_Sadiq/my_will/backends/io_adapters/base.py", line 43, in handle_incoming_event
    m = self.normalize_incoming_event(event)
  File "/Users/Sameer_Sadiq/my_will/backends/io_adapters/hipchat.py", line 622, in normalize_incoming_event
    channel = clean_for_pickling(self.channels[event["xmpp_jid"]])
KeyError: u'237466_asktherunbot@conf.hipchat.com'
```

**Comment by skoczen** [comment]
Hey @sameersadiq1 - this suggests that the list of channels/rooms isn't updated.

A few followups - 
1) What version of will?
2) How recently was this room added?
3) What is the code (i.e. the `say()`/`reply()` that's triggering this?

Thanks!

**Comment by sameersadiq1** [comment]
> Hey @sameersadiq1 - this suggests that the list of channels/rooms isn't updated.

Hasn't been updated where?

1. I encountered this issue using 2.0.
2. This room has been in existence for a few months now. I used it all the time using the 1.0.x versions.
3. I think it's trying to process the incoming message(@respond_to, @hear). It's only throwing this error when it's coming from a group.

**Comment by skoczen** [comment]
Updated - in Will's internal list of rooms.

1. Version - can you be specific?  `@will version` will get it to you.
2. Great.
3. Can you paste the actual plugin code, or at least the line that's causing the error?  

Thanks!

**Comment by sameersadiq1** [comment]
1 it says 1.0.2 when i type in will version
3.  Is this not the line of code you're looking for? 
`channel = clean_for_pickling(self.channels[event["xmpp_jid"]])`
I get the same error for any message i send with this same message. I've tried this with multiple groups as well.  
Here's what I'm trying will to respond to
```
@respond_to("^(?:hi|hey)$")
    def hi(self, message):
        """hi: I know how to say hello!"""
        self.reply("hello!")

```
and here's the error

```
Mon, 20 Nov 2017 11:11:17 [CRITICAL] Error handling incoming event {'body': '@SameerSadiq hi', 'xmpp_jid': u'237466_room_for_one@conf.hipchat.com', 'from': '237466_room_for_one@conf.hipchat.com/Jenkins', 'parent': None, '_index': 0, 'loaded_plugins': set([]), 'namespace': u'jabber:client', 'mid': 'e777c3ff-3bd0-422f-b9f1-0071af072efb', 'ts': '1511194030.330266', 'to': '237466_4376641@chat.hipchat.com/bot||proxy|pubproxy-b600.hipchat.com|5232', 'tag': u'{jabber:client}message', 'plugins': OrderedDict(), 'iterables': [], 'type': 'groupchat'}:
Traceback (most recent call last):
  File "/Users/Sameer_Sadiq/will/backends/io_adapters/base.py", line 43, in handle_incoming_event
    m = self.normalize_incoming_event(event)
  File "/Users/Sameer_Sadiq/will/backends/io_adapters/hipchat.py", line 622, in normalize_incoming_event
    channel = clean_for_pickling(self.channels[event["xmpp_jid"]])
KeyError: u'237466_room_for_one@conf.hipchat.com'
```

**Comment by skoczen** [comment]
Hey @sameersadiq1 - so, I just want to confirm, because it makes a big difference.

Version 1.0.2 is the old version, and 2.0 is what's out in beta right now (to leave beta this week.)   The version response you have makes me worry a bit - your Will says it's 1.0.2, but the code you're pasting from your error is 2.x.

What's `pip list` say?  Are you running in a virtualenv?  Any un-killed python threads running in the background?

The second bit of response code was what I was looking for - thanks1  If you're on v1, you'll need to pass in the `message` object (i.e. `self.reply("hello", message=message)`)  On v2, this isn't needed, and the documentation isn't versioned right now, so perhaps that's the confusion!

Let me know where that leaves you, and we'll dig in further!

**Comment by sameersadiq1** [comment]
will version
Will: I'm running version 2.1.0

still get the same error when sending a message in a group

**Comment by wontonst** [comment]
This issue appears fixed in #340 please reopen if that's not the case.
